modules/core/src/main/scala/org/tesselation/domain/cluster/storage/AddressStorage.scala

```diff
@@ -1,12 +1,21 @@
 package org.tesselation.domain.cluster.storage
 
+import org.tesselation.infrastructure.db.schema.StoredAddress
 import org.tesselation.schema.address.Address
 import org.tesselation.schema.balance.Balance
 
+sealed trait DBUpsertAction[A] { val value: A }
+case class Insert[A](value: A) extends DBUpsertAction[A]
+case class Update[A](value: A) extends DBUpsertAction[A]
```

Comment:
implementation leak. it can be stored in memory as well. you can remove `db` and keep it just as `UpsertAction` probably.

Reply:
Refactored.